coresimd/x86/sse2.rs

```diff
@@ -1145,7 +1145,7 @@ pub unsafe fn _mm_setzero_si128() -> __m128i {
 )]
 #[stable(feature = "simd_x86", since = "1.27.0")]
 pub unsafe fn _mm_loadl_epi64(mem_addr: *const __m128i) -> __m128i {
-    _mm_set_epi64x(0, simd_extract((*mem_addr).as_i64x2(), 0))
+    _mm_set_epi64x(0, simd_extract(ptr::read_unaligned::<__m128i >(mem_addr).as_i64x2(), 0))
```

Comment:
The type annotation here (`::<__m128i>`) shouldn't be necessary. 

Maybe one can just use `_mm_loadu_si128`  here ?

Reply:
Depends - in my case, I need to do a shuffle afterwards anyway, so it comes out to be about the same. If you just wanted to load the lower 64 bits though, then adding in an unnecessary shuffle afterwards would likely cost some performance. At any rate, the fact that the instruction, when seemingly used correctly, yields a segfault is fairly surprising.

Reply:
The `ptr::read_unaligned` reads 128bits, and so does `_mm_loadu_si128`, or what am I missing?

Reply:
Oh, yeah, you're right. I actually don't think I know how to write this PR, now that I think about it.

Reply:
You are doing great! I think you can replace the `ptr::read_unaligned(mem_addr)` with just `_mm_loadu_si128(mem_addr)` and that should just work.